src/main/java/com/venafi/vcert/sdk/connectors/tpp/ZoneConfiguration.java

```diff
@@ -74,47 +93,14 @@ public void updateCertificateRequest(CertificateRequest request) {
                                     }
                                 }
                                 if(!sizeOK) {
-                                    List<Integer> reversedKeySizes = new ArrayList<>(keyConf.keySizes()); // not reversing the original
-                                    reversedKeySizes.sort(Collections.reverseOrder());
-                                    request.keyLength(reversedKeySizes.get(0));
+                                    request.keyLength(keyConf.keySizes().get(0));
```

Comment:
Seems like there is a change in the logic. Before we were taking last size (probably the biggest?) while now we take the first. Is this desirable?

Reply:
TPP policy currently only allows a single key size which means there was no difference in the end result between the old and new code.  If TPP changes in the future we want to default to the smallest key size allowed by policy because it will be the most compatible while still complying with enterprise security policy.  For example, it would most likely be 2048 and 4096 today and the appropriate default between those is 2048.